### PR TITLE
.github: run system tests on arm64

### PR DESCRIPTION
**Pull request by evgeniiz321**

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -96,6 +99,8 @@ jobs:
         run: |
           make all
           echo "$(pwd)/bin" >> $GITHUB_PATH
+        env:
+          GOARCH: ${{ matrix.os.arch }}
```
Comment (roman-khimov) [inline .github/workflows/run-tests.yml:103]: Shouldn't be needed.

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -121,23 +126,15 @@ jobs:
         with:
           repo: 'nspcc-dev/neofs-rest-gw'
           version: 'tags/v0.7.2'
-          file: 'neofs-rest-gw-linux-amd64'
+          file: 'neofs-rest-gw-linux-${{ matrix.os.arch }}'
           target: 'neofs-testcases/neofs-rest-gw'
 
-      - name: Download latest stable neofs-http-gw
```
Comment (roman-khimov) [inline .github/workflows/run-tests.yml:127]: Worth a separate commit.

**Comment by evgeniiz321** [comment]
@roman-khimov can you please take a look at these logs:
[hang_logs.zip](https://github.com/nspcc-dev/neofs-node/files/15118997/hang_logs.zip)

This command hangs infinitely:

```
[MainThread] 2024-04-25 13:59:41 [INFO] Executing command: ./neofs-s3-authmate --debug --with-log --timeout 1m issue-secret --wallet /Users/evgeniiz/Projects/neofs-testcases-nspcc/neofs-testcases/TemporaryDir/3ec9a0e3-cd46-4700-ac17-1a867a382f5f.json --gate-public-key=028c56444d899e8015b82faba961da97ac18afd5fe0d7ff09779f238dbf9e111c5 --peer localhost:52889 --container-friendly-name 8e05fbcf-04c4-41e3-99c0-69c87cc7783f --bearer-rules /Users/evgeniiz/Projects/neofs-testcases-nspcc/neofs-testcases/pytest_tests/data/s3_bearer_rules.json --container-placement-policy 'REP 1' --container-policy pytest_tests/data/container_policy.json
```

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (evgeniiz321) [reply .github/workflows/run-tests.yml]: Removed

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (evgeniiz321) [reply .github/workflows/run-tests.yml:127]: Done

File: .github/workflows/run-tests.yml [hunk 3]
```diff
@@ -244,13 +252,13 @@ jobs:
       - name: Post the link to the report
         id: post_report_link
         timeout-minutes: 60
-        if: always() && steps.put_report.outcome == 'success'
+        if: always() && steps.put_report.outcome
```
Comment (roman-khimov) [inline .github/workflows/run-tests.yml:255]: Why? If it fails this link is irrelevant.

File: .github/workflows/run-tests.yml [hunk 4]
```diff
@@ -98,44 +116,34 @@ jobs:
...
-          target: 'neofs-testcases/neofs-s3-gw'
-
-      - name: Download latest stable neofs-s3-gw-authmate
-        uses: dsaltares/fetch-gh-release-asset@1.1.1
-        with:
-          repo: 'nspcc-dev/neofs-s3-gw'
-          version: 'tags/v0.30.0'
-          file: 'neofs-s3-authmate-linux-amd64'
-          target: 'neofs-testcases/neofs-s3-authmate'
+      - name: Build neofs-s3-gw
```
Comment (roman-khimov) [inline .github/workflows/run-tests.yml:119]: You don't mention this change in the commit message.

File: .github/workflows/run-tests.yml [hunk 5]
```diff
@@ -98,44 +116,34 @@ jobs:
...
+        working-directory: neofs-s3-gw
 
-      - name: Download latest stable neofs-rest-gw
-        uses: dsaltares/fetch-gh-release-asset@1.1.1
-        with:
-          repo: 'nspcc-dev/neofs-rest-gw'
-          version: 'tags/v0.7.2'
-          file: 'neofs-rest-gw-linux-amd64'
-          target: 'neofs-testcases/neofs-rest-gw'
+      - name: Build neofs-rest-gw
```
Comment (roman-khimov) [inline .github/workflows/run-tests.yml:126]: As well as this one.

Comment (evgeniiz321) [reply .github/workflows/run-tests.yml:126]: Split into different commits.

File: .github/workflows/run-tests.yml (hunk 4 above)
Comment (evgeniiz321) [reply .github/workflows/run-tests.yml:119]: Split into different commits.

File: .github/workflows/run-tests.yml (hunk 3 above)
Comment (evgeniiz321) [reply .github/workflows/run-tests.yml]: My mistake. Reverted it back.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**